src/web_interface/templates/dependency_graph.html

```diff
@@ -0,0 +1,351 @@
+{% extends "base.html" %}
+
+{% set active_page = "Database" %}
+
+{% block head %}
+    <title>Dependency Graph</title>
+    <style type="text/css">
+        #dependencyGraph {
+            height: 800px;
+        }
+
+        #dependencyGraphLegend {
+            height: 500px;
+        }
+    </style>
+    <script src="{{ url_for('static', filename='vis/vis-network.js') }}"></script>
+    <link href="{{ url_for('static', filename='vis/dist/vis-network.min.css') }}" rel="stylesheet" type="text/css" />
+{% endblock %}
+
+
+{% block body %}
+    <div class="modal" id="context-menu" tabindex="-1" role="dialog" aria-labelledby="contextLabel" aria-hidden="true">
+        <div class="modal-dialog modal-dialog-centered" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="contextLabel">Additional information</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    Node:
+                    <p id='node-link'>No node selected.</p>
+                    <br>
+                    File type:
+                    <p id='file-type'>No node selected.</p>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="header mb-4" style="word-wrap: break-word">
+        <h3>
+            Dependency Graph for File {{ uid | replace_uid_with_hid }}
+            <br>
+            <span style="font-size: 15px"><strong>UID:</strong> {{ uid }}</span>
+        </h3>
+    </div>
+
+
+
+
+    <div class="row mb-3">
+        <div class="col-md-8">
+            <div id="dependencyGraph"></div>
+        </div>
+        <div class="col-md-4">
+            <div class="header mb-4" style="word-wrap: break-word">
+                <h4 style="text-align: center">Legend</h4>
+            </div>
+            <div id="dependencyGraphLegend"></div>
+            <span style="font-size: 15px">
+                A context menu providing additional information about the nodes is available on right click.
+            </span>
+        </div>
+    </div>
+
+<!--  insert js graph magic here  -->
+<script>
+    var nodes, groups, edges, network, legend_network, allNodes;
+    var highlightActive = false;
+
+    // convenience method to stringify a JSON object
+    function toJSON(obj) {
+        return JSON.stringify(obj, null, 4);
+    }
+
+    var get_nodes = eval( {{ nodes | safe }} )
+    var get_edges = eval( {{ edges | safe }} )
+    var get_groups = eval( {{ groups | safe }} )
```

Comment:
The `eval()` shouldn't be necessary (and we generally want as few `eval()`s in our code as possible :sweat_smile: )
```suggestion
    var get_nodes = {{ nodes | safe }}
    var get_edges = {{ edges | safe }}
    var get_groups = {{ groups | safe }}
```